homeassistant/components/honeywell/__init__.py

```diff
@@ -134,73 +131,13 @@ def __init__(
         self._password = password
         self.devices = devices
 
-    async def _retry(self) -> bool:
-        """Recreate a new somecomfort client.
-
-        When we got an error, the best way to be sure that the next query
-        will succeed, is to recreate a new somecomfort client.
-        """
-        self._client = await self._hass.async_add_executor_job(
-            get_somecomfort_client, self._username, self._password
-        )
-
-        if self._client is None:
-            return False
+    async def retry_login(self) -> bool:
+        """Fire of a login retry."""
 
-        refreshed_devices = [
-            device
-            for location in self._client.locations_by_id.values()
-            for device in location.devices_by_id.values()
-        ]
-
-        if len(refreshed_devices) == 0:
-            _LOGGER.error("Failed to find any devices after retry")
+        try:
+            await self._client.login()
+        except AIOSomecomfort.SomeComfortError:
+            await asyncio.sleep(UPDATE_LOOP_SLEEP_TIME)
             return False
 
-        for updated_device in refreshed_devices:
-            if updated_device.deviceid in self.devices:
-                self.devices[updated_device.deviceid] = updated_device
-            else:
-                _LOGGER.info(
-                    "New device with ID %s detected, reload the honeywell integration"
-                    " if you want to access it in Home Assistant"
-                )
-
-        await self._hass.config_entries.async_reload(self._config.entry_id)
         return True
-
-    async def _refresh_devices(self):
-        """Refresh each enabled device."""
-        for device in self.devices.values():
-            await self._hass.async_add_executor_job(device.refresh)
-            await asyncio.sleep(UPDATE_LOOP_SLEEP_TIME)
```

Comment:
This whole thing was bad - when multiple devices are setup, the integration was updating every device for each device async_update! (3 devices = 9 updates)
I've moved the update per device into the climate async_update.